Use Autoconf to check for strlcpy

ecerulm opened the pull request:
Autoconf can check for strlcpy. If it finds it without having to use
a library it won't add anything to LIBS. Otherwise it will try
libbsd and if finds it there it will link libbsd.

Use also AC_CHECK_HEADERS to only include bsd/string.h if it actually
exists.

calleerlandsson commented on configure.ac:
```diff
@@ -9,6 +9,14 @@ AC_TYPE_SSIZE_T
 AC_CHECK_FUNCS([memmove setlocale strdup])
+AC_SEARCH_LIBS([strlcpy], [bsd], [], [
+  AC_MSG_ERROR([unable to find strlcpy() function])
+])
```
If neither `strlcpy` nor libbsd is present on the system, will this error out? #10 bundles implementations of `strlcpy` and `strlcat` which are used if `AC_CHECK_FUNCS` doesn't find the functions. Would that do as a replacement for `AC_SEARCH_LIBS`?

ecerulm replied:
Yes it will error out,  I guess the message should be "unable to find strlcpy(). Make sure libbsd is installed"  

Guess if #10 gets merged you can remove the whole `AC_SEARCH_LIBS`  since you are not dependent on `libbsd`  after #10. 

This was more  to solve the current issue of warning the user that there was no `strlcpy` at `./configure`  instead of at compile time. It depends on when are you going to merge #10, you can always remove this `AC_SEARCH_LIBS` after merging #10.

calleerlandsson commented on src/ui.c:
```diff
@@ -45,7 +50,7 @@ start_curses()
 
 	signal(SIGINT, int_handler);
-	freopen("/dev/tty", "r", stdin);
+	assert(freopen("/dev/tty", "r", stdin));
```
Would using an `if` statement to check the return value and `err` to error if `freopen` returns `NULL` improve consistency with other parts of the code base?

calleerlandsson commented:
Thanks for contributing @ecerulm! I'm sorry for not getting back to you earlier. I've been travelling without internet access for a month.

I left some line comments.

ecerulm commented:
I updated the error message if `strlcpy` cannot be found prompting the user to install `libbsd`. #10 will make that obsolete when it's merged but until that happens it's needed to build in Linux. 

I replaced the `assert` with `if/err` construct.

calleerlandsson commented:
Thanks @ecerulm. I will squash and merge this.

calleerlandsson commented:
Merged as 59c434f1ecc78ccfd6ba53de37f7ef9b6387624a and 0c03b0d76a8546016f824f7f977593fcbf1dab90.
